packages/ckeditor5-utils/src/observablemixin.js

```diff
@@ -258,11 +261,33 @@ const ObservableMixin = {
 		this[ methodName ] = function( ...args ) {
 			return this.fire( methodName, args );
 		};
+
+		this[ methodName ][ _decoratedOriginal ] = originalMethod;
+
+		if ( !this[ _decoratedMethods ] ) {
+			this[ _decoratedMethods ] = [];
+		}
+
+		this[ _decoratedMethods ].push( methodName );
 	}
 };
 
 extend( ObservableMixin, EmitterMixin );
 
+// Override the EmitterMixin stopListening method to be able to clean decorated methods.
+ObservableMixin.stopListening = function( emitter, event, callback ) {
```

Comment:
This is a sidetrack that I noticed in the previous attempt at bubbling events. Some tests started to fail because of `ModelTestEditor` is destroying a part of the editor and it causes that all events stop firing (including those for decorated methods so there is no way to call the original method). BTW this can be removed from this PR because it is no longer causing a problem.

Reply:
1.  Have `x.foo()` decorated
2.  Call `x.stopListening()`
3.  Call `x.foo()`. Problem: nothing happens (the original `foo()` method is not executed)

While not super popular scenario, it's possible. So it'd be good to have it covered. I'd keep this code, but we need to have it explained in the code clearly why this is needed.